From 17e13be56a17e63dbbeb7cd9d6dfeb1e222bedd6 Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Sun, 6 Jul 2025 20:05:42 +0200
Subject: [PATCH 008/115] dt-bindings: reset: mediatek: Add support for MT6785

Add compatibles and bindings for all reset controllers found in MT6785 SoC.

Signed-off-by: Dinolek <git@dinolek.me>
---
 .../reset/mediatek,mt6785-resets.h            | 31 +++++++++++++++++++
 1 file changed, 31 insertions(+)
 create mode 100644 include/dt-bindings/reset/mediatek,mt6785-resets.h

diff --git a/include/dt-bindings/reset/mediatek,mt6785-resets.h b/include/dt-bindings/reset/mediatek,mt6785-resets.h
new file mode 100644
index 000000000..512c675cb
--- /dev/null
+++ b/include/dt-bindings/reset/mediatek,mt6785-resets.h
@@ -0,0 +1,31 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+
+#ifndef _DT_BINDINGS_RESET_CONTROLLER_MT6785
+#define _DT_BINDINGS_RESET_CONTROLLER_MT6785
+
+#define MT6785_INFRA_THERM_CTRL_RST		0
+#define MT6785_INFRA_UFS_HCI_RST		1
+#define MT6785_INFRA_PMIC_WRAP_RST		2
+#define MT6785_INFRA_UFS_UNIPRO_RST		3
+#define MT6785_INFRA_SVS_RST			4
+#define MT6785_INFRA_UFS_AES_RST		5
+
+#define MT6785_MMSYS_DISP_OVL0_RST		13
+#define MT6785_MMSYS_DISP_OVL0_2L_RST		14
+#define MT6785_MMSYS_DISP_DSI0_RST		25
+
+#define MT6785_TOPRGU_MM_RST			1
+#define MT6785_TOPRGU_MFG_RST			2
+#define MT6785_TOPRGU_VENC_RST			3
+#define MT6785_TOPRGU_VDEC_RST			4
+#define MT6785_TOPRGU_IMG_RST			5
+#define MT6785_TOPRGU_MD_RST			7
+#define MT6785_TOPRGU_CONN_RST			9
+#define MT6785_TOPRGU_CONN_MCU_RST		12
+#define MT6785_TOPRGU_IPU0_RST			14
+#define MT6785_TOPRGU_IPU1_RST			15
+#define MT6785_TOPRGU_AUDIO_RST			17
+#define MT6785_TOPRGU_CAMSYS_RST		18
+#define MT6785_TOPRGU_RST_NUM			19
+
+#endif  /* _DT_BINDINGS_RESET_CONTROLLER_MT6785 */
-- 
2.56.0

